pandas/io/common.py

```diff
@@ -201,6 +201,33 @@ def _validate_header_arg(header):
                         "header=int or list-like of ints to specify "
                         "the row(s) making up the column names")
 
+def _stringify_path(filepath_or_buffer):
+    """Return the argument coerced to a string if it was a pathlib.Path
+       or a py.path.local.LocalPath
+
+    Parameters
+    ----------
+    filepath_or_buffer : object to be converted
+
+    Returns
+    -------
+    str_filepath_or_buffer : a the string version of the input path
+    """
+    try:
+        import pathlib
+        if isinstance(filepath_or_buffer, pathlib.Path):
+            return text_type(filepath_or_buffer)
+    except:
+        pass
+    try:
+        from py.path.local import LocalPath
```

Comment:
hmm, I don't think so, if the import is NOT available, then its tested each time, yes? (if its already their then yes this is a simple dict lookup). Since this affects everyone on py2 on every import this *might* be excessive (though against the cost of a read_csv its trivial).